### PR TITLE
Refactors for grinding & juicing

### LINKED ISSUE #87719: Active industrial slime extracts are destroyed by chemical grinder using conveyor belt setup
### Client Version:

515.1643

### Issue Summary:

Industrial slime extracts are destroyed by chem grinder when they have been injected with plasma and are actively producing items. Looking back at TG discord logs, this has been an issue for at least a year.

### Round ID:

[241433](https://scrubby.melonmesa.com/round/241433)

### Test Merge Information:

1. Inject industrial (dark purple) extract with plasma
2. Use conveyor belt to move active industrial extract into grinder

### Reproduction Steps:

1. Inject plasma into industrial (dark purple) extract
2. Use conveyor belt to slide extract to chem grinder

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request
- Grinding & juicing are now recursive operations (grind the item & all its contents) not just for reagent grinder but also for mortar pedestal & plumbing grinder so you can get all them juices
- Fixes #87719. Plumbing grinder won't destroy slime extracts after grinding
- Mortar pedestal now grinds & juices more items since it correctly checks if that object has more reagents to offer
- Reagent grinder, plumbing grinder & mortar pedestal now respect `blend_requirements()` so you get consistent behaviour across them

## Changelog
:cl:
fix: mortar pedestal now grinds & juices items that previously could not be processed
fix: plumbing grinder won't destroy slime extracts after grinding
refactor: grinding & juicing code has been refactored overall. Please report bugs on github
/:cl:

### CODE REVIEW

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -1027,36 +1027,93 @@
...
-		return FALSE
+		return
+
+	. = juice_atom(target_holder, user)
+
+	//reccursive juicing to get all them juices
+	var/result
+	for(var/obj/item/ingredient as anything in get_all_contents_type(/obj/item))
+		if(ingredient == src)
+			continue
```
Comment (Time-Green) [inline code/game/objects/items.dm:1091]: Kinda ditto here

**Review by Time-Green — CHANGES_REQUESTED** [review]
Looks good! Question though

File: code/game/objects/items.dm [hunk 2]
```diff
@@ -1027,36 +1027,93 @@
...
+	if(on_grind() == -1 || target_holder.holder_full())
 		return
 
+	. = grind_atom(target_holder, user)
+
+	//reccursive grinding to get all them juices
+	var/result
+	for(var/obj/item/ingredient as anything in get_all_contents_type(/obj/item))
+		if(ingredient == src)
+			continue
```
[the remark this answers is not in the document]
Comment (SyncIt21) [reply code/game/objects/items.dm:1048]: That's how the proc works. It first returns itself and then it's contents it does not mean that the object is inside itself.

File: code/game/objects/items.dm (hunk 1 above)
Comment (SyncIt21) [reply code/game/objects/items.dm:1091]: Same as above answer. That's how the proc structures it's list

**Comment by SyncIt21** [comment]
Hi :)

**Review by Time-Green — APPROVED** [review]

**Pull request merged**